amptorch/descriptor/Gaussian/__init__.py

```diff
@@ -32,25 +36,32 @@ def __init__(self, Gs, elements, cutoff_func="cosine", gamma=None):
         self.get_descriptor_setup_hash()
 
     def prepare_descriptor_parameters(self):
-        self.descriptor_setup = {}
-        for element in self.elements:
-            if element in self.Gs:
-                self.descriptor_setup[
-                    element
-                ] = self._prepare_descriptor_parameters_element(
-                    self.Gs[element], self.element_indices
-                )
-            elif "default" in self.Gs:
-                self.descriptor_setup[
-                    element
-                ] = self._prepare_descriptor_parameters_element(
-                    self.Gs["default"], self.element_indices
-                )
-            else:
-                raise NotImplementedError(
-                    "Symmetry function parameters not defined properly"
-                )
-
+        if isinstance(self.Gs, dict):
+            self.descriptor_setup = {}
+            for element in self.elements:
+                if element in self.Gs:
+                    self.descriptor_setup[
+                        element
+                    ] = self._prepare_descriptor_parameters_element(
+                        self.Gs[element], self.element_indices
+                    )
+                elif "default" in self.Gs:
+                    self.descriptor_setup[
+                        element
+                    ] = self._prepare_descriptor_parameters_element(
+                        self.Gs["default"], self.element_indices
+                    )
+                else:
+                    raise NotImplementedError(
+                        "Symmetry function parameters not defined properly"
+                    )
+        elif hasattr(self.Gs, "descriptor_setup"):
+            self.descriptor_setup = self.Gs.descriptor_setup
```

Comment:
To verify, in the custom case `self.Gs` is a `GaussianDescriptorSet` object, correct?

Reply:
yes, I left it as this `hasattr` instead of `isinstance` in case I came up with a different, additional system like the `GDS` so the same attribute could be used